sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -555,6 +555,11 @@ RT::PiProgram ProgramManager::getBuiltPIProgram(
     return BuiltProgram.release();
   };
 
+  // FIXME: Level Zero creates multiple PiDevices for a single physical device
+  // when sub-device is partitioned into sub-sub-devices. Sub-sub-device is
+  // technically a command queue and we should not build program for each
+  // command queue. PiDevice is probably not the right abstraction for a Level
+  // Zero command queue.
```

Comment:
This comment should be moved to L0 plugin (probably around https://github.com/intel/llvm/blob/64e92cbc41d1bcde9c728798e4e1fed9e3fab253/sycl/plugins/level_zero/pi_level_zero.cpp#L2188)

The sub-sub device is a L0 sub-device handle + ordinal/index to the specific CCS.
Representing it as a PI device makes sense in terms of hierarchy, and also with a vision that L0 might add true sub-sub-device handles in the future.

For the purpose of this PR I think the `PI_DEVICE_INFO_BUILD_ON_SUBDEVICE` returning `false` should also mean that the sub-sub-devices (and you can easily check that at SYCL RT level) can re-use root-device builds.